backend/src/core/models/game.js

```diff
@@ -93,19 +108,124 @@ export class Game {
     });
   }
 
-  updateGameDetails(newDevelopers, newGenres, newDescription) {
-    if (newDevelopers.length !== 0 && this.developers.length === 0)
-      this.developers = Array.from(newDevelopers);
+  getReleaseDate(page) {
+    const releaseDateElement = page.querySelector(".release_date .date");
+
+    if (!releaseDateElement) return "";
+
+    const releaseDate = new Date(`${releaseDateElement.textContent.trim()} UTC`);
+
+    return releaseDate == "Invalid Date" ? "" : releaseDate;
+  }
+
+  getDevelopers(page) {
+    const developers = page.querySelector(".dev_row #developers_list");
+
+    if (!developers) return [];
+
+    return Array.from(developers.children).map((developer) =>
+      developer.textContent.trim(),
+    );
+  }
+
+  getGenres(page) {
+    const genres = page.querySelector("#genresAndManufacturer span");
+
+    if (!genres) return [];
+
+    return Array.from(genres.children)
+      .map((genre) => genre.textContent.trim())
+      .filter((genre) => !!genre);
+  }
+
+  getDescription(page) {
+    const description = page.querySelector(".game_description_snippet");
+
+    if (!description) return "";
+
+    return description.textContent.trim();
+  }
+
+  updateGameDetails(page) {
+    this.updateDevelopers(page);
+    this.updateGenres(page);
+    this.updateDescription(page);
+  }
+
+  updateDevelopers(page) {
+    if (this.developers.length !== 0) return;
+
+    this.developers = this.getSteamDbDevelopers(page);
```

Comment:
I would inline this method. Same below.

Reply:
I decided to skip this, I believe it's much better readability if I don't inline it. Not sure you will even re-check this PR but if you do could you explain why it's better? I can change it np.